crates/codegen/runtime/generator/src/built_ins/mod.rs

```diff
@@ -0,0 +1,97 @@
+use std::path::Path;
+use std::rc::Rc;
+
+use anyhow::Result;
+use codegen_language_definition::model::{
+    BuiltIn, BuiltInField, BuiltInFunction, BuiltInType, Language,
+};
+use infra_utils::codegen::CodegenFileSystem;
+use semver::Version;
+
+pub fn render_built_ins(
+    file_system: &mut CodegenFileSystem,
+    language: &Rc<Language>,
+    output_dir: &Path,
+    render_fn: impl Fn(&[BuiltIn]) -> String,
+) -> Result<()> {
+    let versions = language.collect_built_ins_versions();
+    let file_extension = language.file_extension.clone().unwrap_or_default();
+    for version in &versions {
+        let built_ins = filter_built_ins_for_version(&language.built_ins, version);
+        let contents = render_fn(&built_ins);
+
+        let output_path = output_dir.join(format!("{version}{file_extension}"));
+        std::fs::write(&output_path, contents)?;
+        file_system.mark_generated_file(output_path)?;
```

Comment:
is there a reason we are not using `file_system.write_file()`? It will:

- mark the file as generated
- add the `// header` warning
- check if it is up to date in CI.

The built-in `std::fs::write()` doesn't.

Reply:
I thought about this but it would mean that we need to update the generators to handle Solidity files explicitly. I know Slang is mostly about Solidity right now, but it would mean losing some generality. I reverted to using `fs::write()` when I got the error from infra saying that it didn't know how to generate the header.

Reply:
I think this is fine to add in `infra_utils` (`// XXX` header, and no formatting). It handles 10+ other languages, and is not specific to the target language Slang is generating, but handles other assets/snapshots/docs/etc...